homeassistant/components/songpal/media_player.py

```diff
@@ -212,13 +212,18 @@ def name(self):
     @property
     def unique_id(self):
         """Return a unique ID."""
-        return self._sysinfo.macAddr
+        return self._sysinfo.macAddr or self._sysinfo.wirelessMacAddr
 
     @property
     def device_info(self) -> DeviceInfo:
         """Return the device info."""
+        connections = set()
+        if self._sysinfo.macAddr:
+            connections.add((dr.CONNECTION_NETWORK_MAC, self._sysinfo.macAddr))
+        if self._sysinfo.wirelessMacAddr:
+            connections.add((dr.CONNECTION_NETWORK_MAC, self._sysinfo.wirelessMacAddr))
```

Comment:
Yeah I saw how that failed. Turns out I was testing a single case instead of the whole unit, oops!

I changed the test to not use a mock at all for SysInfo. This makes it more reliable overall.